src/storage.rs

```diff
@@ -609,6 +609,52 @@ pub async fn read_subscribed_channels(peer_id: &str) -> Result<Vec<String>, Box<
     Ok(channels)
 }
 
+/// Get all available channels (discovered from the network)
+pub async fn read_available_channels() -> Result<Channels, Box<dyn Error>> {
```

Comment:
If this is the same as readchannels why not use readchannels rather than create a wrapper around it.